src/lib/db/transaction.ts

```diff
@@ -81,6 +86,25 @@ export function withTransactional<S>(
     return service;
 }
 
+export function withRollback<S>(
+    serviceFactory: (db: Knex) => S,
+    db: Knex,
+): WithRollback<S> {
+    const service = serviceFactory(db) as WithRollback<S>;
+
+    service.rollback = async <R>(fn: (service: S) => R) => {
+        const trx = await db.transaction({ isolationLevel: 'serializable' });
```

Comment:
The highest possible isolation level, is good for keeping consistency but it might backfire in terms of performance and resource locking and reducing the concurrency level.

It's true we're not going to use it that frequently, but the moment it's used it might have a negative impact in the system performance overall.

Reply:
that's why I'm considering using 'read committed' which is postgres default. I'm starting with the most strict level for the experimentation purposes. 

I can give the context where it's gonna be used and why the tx may take a few seconds:
* you open transaction
* you apply your change request
* you run playground evaluation (may take a few seconds so bad for other transactions)
* you roll back the change

Reply:
Spending a few seconds worries me with the current isolation level, but experimenting seems fine, I'd try to see if we suddenly spawn up a lot of new pods when actively hitting this functionality while at the same time we have some load on the client endpoints and maybe the UI.

Perhaps read committed works well because it prevents dirty reads and that's all we really want here, right?

Reply:
This is sth behind a flag so I'll do internal testing first and probably downgrade the isolation level